### ISSUE TITLE
[HttpTrigger] Multiple function execution for the same request when scaling out

### ISSUE THREAD

**Issue by georgechond94** [Azure/azure-functions-dotnet-worker#826]
When an [HttpTrigger] function scales out because of a burst of concurrent requests, the same request is being picked up by multiple instances at the same time, resulting in processing the same request multiple times.
The number of executions is quite random it seems, it depends on when the function will scale out.

Example with **10** concurrent HTTP requests: 

Maximum Scale Out Limit to 200 (default):

![image](https://user-images.githubusercontent.com/6044597/157682887-755b2887-6473-49d7-b12e-cdb9eb4b1d26.png)

We can see that whenever it scales out and there are still in-progress requests there are always some requests that are being handled by multiple instances at the same time. (1st spike) 
Whenever the scaling out is done everything is back to normal (2nd spike)
(PS. I know it scales out as I am getting the "Application Started" logs in between of the HTTP invocation logs)

After setting the Maximum Scale Out Limit to 1 everything works as expected always:
![image](https://user-images.githubusercontent.com/6044597/157646899-c42aaca6-4d41-4401-a924-7084c781f288.png)

.NET 6
Functions v4 (Isolated)
Dependencies:
```
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.13" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.3.0" OutputItemType="Analyzer" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.6.0" />
```

I can provide a minimum repro if needed.

**Comment by kshyju** [comment]
Hi @georgechond94 👋 How are you determining it is the same request being served by multiple instances? Are you seeing http invocation with same [InvocationId ](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.functions.worker.functioncontext.invocationid?view=azure-dotnet)being served from multiple worker instances? Curious to know what data points you used to reach this conclusion.  Not really able to make any conclusions with the graph image posted above.

**Comment by georgechond94** [comment]
Hi @kshyju, thanks for your reply! Checked the logs and the `azurefunctions_invocationid` is always different.
Basically the body I am POSTing is unique in every request. 
The behavior I am seeing in the logs is that some payloads are mentioned twice, and this especially happens after an `Application started. Press Ctrl+C to shut down.` message which I guess means the function scaled out.

If the function is warm (and scaled out already recently) or I set the Maximum Scale Out Limit to 1 and send the same 10 requests, they are always being processed exactly 10 times. 
Thanks!
